File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import java.util.List;

import com.pinyougou.pojogroup.Cart;

public interface CartService {

	 List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);
	 
	 List<Cart> findCartListFromRedis(String username);
	 
	 void saveCartListToRedis(List<Cart> cartList,String username);
	 
	 List<Cart> mergeCartList(List<Cart> cartList1,List<Cart> cartList2);
	 
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService',function($http){
	
	this.findCartList=function(){
		return $http.get('cart/findCartList.do');
	}
	
	this.addGoodsToCartList=function(itemId,num){
		return $http.get('cart/addGoodsToCartList.do?itemId='+itemId+'&num='+num);
	}
	
	
	this.sum=function(cartList){
		var totalValue = {totalNum:0,totalMoney:0};
		for(var i=0;i<cartList.length;i++){
			var items = cartList[i].orderItemList;
			for(var j=0;j<items.length;j++){
				totalValue.totalNum+=items[j].num;
				totalValue.totalMoney+=items[j].totalFee;
			}
		}
		return totalValue;
	}
})